File: libsignal-service/src/main/java/org/whispersystems/signalservice/api/websocket/HealthMonitor.java
package org.whispersystems.signalservice.api.websocket;

/**
 * Callbacks to provide WebSocket health information to a monitor.
 */
public interface HealthMonitor {
  void onKeepAliveResponse(long sentTimestamp, boolean isIdentifiedWebSocket, boolean keepMonitoring);

  void onMessageError(int status, boolean isIdentifiedWebSocket);
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/websocket/HealthMonitor.kt
package org.whispersystems.signalservice.api.websocket

/**
 * Callbacks to provide WebSocket health information to a monitor.
 */
interface HealthMonitor {
  fun onKeepAliveResponse(sentTimestamp: Long, isIdentifiedWebSocket: Boolean)

  fun onMessageError(status: Int, isIdentifiedWebSocket: Boolean)
}
